tools/header_gate_objects.py

```
@@ -0,0 +1,74 @@
+    r'(?P<name>[A-Za-z_]\w*)'
+    r'(?P<arr>(?:\s*\[[^\]]*\])*)\s*'
+    r'(?P<init>\=\s*[^;{][^;]*|=\s*\{[^;]*\}|\s*\{[^;]*\})\s*;',
+    re.S
+)
+
+def guard_object(src:str, target:str, macro:str):
+    out_lines=[]
+    found=False
+    for line in src.splitlines(True):
+        if 'extern' in line or line.lstrip().startswith('#'):
+            out_lines.append(line); continue
```

Ruff (0.13.1)

Multiple statements on one line (semicolon)

(E702)